### PR TITLE
Reduce # of memcpy from storage engine layer

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/5340/

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/5346/

File: arangod/Transaction/Context.cpp [hunk 1]
```diff
@@ -143,17 +144,26 @@ void transaction::Context::returnStringBuffer(basics::StringBuffer* stringBuffer
...
+  return s;
 }
 
 /// @brief return a temporary std::string object
 void transaction::Context::returnString(std::string* str) {
-  _stdString.reset(str);
+  try {  // put string back into our vector of strings
+    _strings.push_back(str);
+  } catch (...) {
+    // no harm done. just wipe the builder
```
Comment (jsteemann) [inline arangod/Transaction/Context.cpp]:
```suggestion
    // no harm done. just wipe the string
```

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/5360/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/5362/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3965/

**Pull request merged**